yarn-project/cli-wallet/src/cmds/index.ts

```diff
@@ -611,5 +604,91 @@ export function injectCommands(program: Command, log: LogFn, debugLogger: DebugL
+    .action(async (address, options) => {
+      const { registerContact } = await import('./register_contact.js');
+      const { from: parsedFromAddress, rpcUrl, secretKey, alias } = options;
+      const client = pxeWrapper?.getPXE() ?? (await createCompatibleClient(rpcUrl, debugLogger));
+      const account = await createOrRetrieveAccount(client, parsedFromAddress, db, secretKey);
+      const wallet = await getWalletWithScopes(account, db);
+
+      await registerContact(wallet, address, log);
+
+      if (db && alias) {
+        await db.storeContact(address, alias, log);
+      }
+    });
+
+  program
+    .command('register-contract')
+    .description("Registers a contract in this wallet's PXE")
+    .argument('[address]', 'The address of the contract to register', address =>
+      aliasedAddressParser('accounts', address, db),
+    )
+    .argument('[artifact]', ARTIFACT_DESCRIPTION, artifactPathParser)
+    .option('--init <string>', 'The contract initializer function to call', 'constructor')
+    .option(
+      '-k, --public-key <string>',
+      'Optional encryption public key for this address. Set this value only if this contract is expected to receive private notes, which will be encrypted using this public key.',
+      parsePublicKey,
+    )
+    .option(
+      '-s, --salt <hex string>',
+      'Optional deployment salt as a hex string for generating the deployment address.',
+      parseFieldFromHexString,
+    )
+    .option('--deployer <string>', 'The address of the account that deployed the contract', address =>
+      aliasedAddressParser('accounts', address, db),
+    )
+    .addOption(createArgsOption(true, db))
+    .addOption(pxeOption)
+    .addOption(createAccountOption('Alias or address of the account to simulate from', !db, db))
+    .addOption(createAliasOption('Alias for the contact. Used for easy reference in subsequent commands.', !db))
+    .action(async (address, artifactPathPromise, _options, command) => {
+      const { registerContract } = await import('./register_contract.js');
+      const {
+        from: parsedFromAddress,
+        rpcUrl,
+        secretKey,
+        alias,
+        init,
+        publicKey,
+        salt,
+        deployer,
+        args,
+      } = command.optsWithGlobals();
+      const client = pxeWrapper?.getPXE() ?? (await createCompatibleClient(rpcUrl, debugLogger));
+      const account = await createOrRetrieveAccount(client, parsedFromAddress, db, secretKey);
+      const wallet = await getWalletWithScopes(account, db);
+
+      const artifactPath = await artifactPathPromise;
+
+      const instance = await registerContract(
+        wallet,
+        address,
+        artifactPath,
+        init,
+        publicKey ? PublicKeys.fromString(publicKey) : undefined,
+        args,
+        salt,
+        deployer,
+        log,
+      );
+
+      if (db && alias) {
+        await db.storeContract(instance.address, artifactPath, log, alias);
+      }
```

Comment:
Should we not be able to register contacts without knowing their contract's artifacts?